base/abstractarray.jl

```diff
@@ -208,6 +208,13 @@ reshape(a::AbstractArray, dims::Int...) = reshape(a, dims)
 
 ## from general iterable to any array
 
+function copy!(dest, src)
+    for x in src
+        push!(dest, x)
```

Comment:
Yes; I did this because `push!` is the only generic "add to collection" function we have. This definition only works if `copy(dest::AbstractArray, src)` is also defined, which it is.

`push!` is key-preserving for non-arrays, but arrays can't do that trick since taking an element out of an array drops the key. Actually this definition can replace `union!` (sets) and `merge!` (dicts). Given that we have copy!, append!, union!, and merge!, it looks quite possible we have too many whole-collection functions and not enough single-element functions. `append!` goes with `push!, but there isn't a function that pairs with `copy!`. Arrays could then implement `copy!`, but not the single-element partner for that function.